include/SDL3/SDL_filesystem.h

```diff
@@ -236,6 +236,127 @@ typedef enum
+    int (SDLCALL * enumerate)(SDL_FSops *fs, const char *path, SDL_EnumerateCallback cb, void *userdata);
+
+    /* delete/unlink/rmdir a path. Will not remove non-empty directories! */
+    int (SDLCALL * remove)(SDL_FSops *fs, const char *path);
+
+    /* rename a file/dir from `oldpath` to `newpath` (there are some standard limitations, documentation forthcoming) */
+    int (SDLCALL * rename)(SDL_FSops *fs, const char *oldpath, const char *newpath);
+
+    /* Create a directory. */
+    int (SDLCALL * mkdir)(SDL_FSops *fs, const char *path);
+
+    /* Obtain basic path metadata. */
+    int (SDLCALL * stat)(SDL_FSops *fs, const char *path, SDL_Stat *stat);
+
+    /* Close the filesystem and free its resources. */
+    void (SDLCALL * closefs)(SDL_FSops *fs);
+
+    /* Used by the SDL_FSops for private data. Don't touch! */
+    void *opaque;
+
+    /* Everything has properties! :) */
+    SDL_PropertiesID props;
+};
+
+
+/*
+ * Create an SDL_FSops that represents the native filesystem, treating `basedir` as
+ * the root of the tree. A NULL basedir is valid and gives you the actual root of the
+ * native filesystem.
+ *
+ * Enumerating root on Windows will give you available drive letters like they were directories,
+ * almost everything else assumes the root is a Unix "/" thing.
+ *
+ * Destroy this when done with SDL_DestroyFilesystem().
+ */
+extern DECLSPEC SDL_FSops * SDLCALL SDL_CreateFilesystem(const char *basedir);
+
+
+/* Use the filesystem... */
+extern DECLSPEC SDL_RWops * SDLCALL SDL_FSopen(SDL_FSops *fs, const char *path, const char *mode);
+extern DECLSPEC int SDLCALL SDL_FSenumerate(SDL_FSops *fs, const char *path, SDL_EnumerateCallback cb, void *userdata);
+extern DECLSPEC int SDLCALL SDL_FSremove(SDL_FSops *fs, const char *path);
+extern DECLSPEC int SDLCALL SDL_FSrename(SDL_FSops *fs, const char *oldpath, const char *newpath);
+extern DECLSPEC int SDLCALL SDL_FSmkdir(SDL_FSops *fs, const char *path);
+extern DECLSPEC int SDLCALL SDL_FSstat(SDL_FSops *fs, const char *path, SDL_Stat *stat);
+
+/* returns `props` field of `context`, creating an SDL_PropertiesID for that field, if it doesn't exist yet. */
+extern DECLSPEC SDL_PropertiesID SDLCALL SDL_GetFSProperties(SDL_FSops *context);
+
+/**
+ * if fs is not NULL, this will call fs->closefs() (if closefs is not NULL), destroy fs->props, and then SDL_free(fs). You can use this
+ * with your own SDL_FSops implementations if that sequence of operations is correct.
+ */
+extern DECLSPEC void SDLCALL SDL_DestroyFilesystem(SDL_FSops *fs);
+
+
+/* some helper functions ... */
+
+/* converts an SDL file timestamp into a win32 FILETIME (100-nanosecond intervals since January 1, 1601). Fills in the two 32-bit values of the win32 struct. */
+extern DECLSPEC void SDLCALL SDL_FileTimeToWindows(Uint64 ftime, Uint32 *low, Uint32 *high);
+
+/* converts an SDL file timestamp into a Unix time_t (seconds since the Unix epoch). */
+extern DECLSPEC Uint64 SDLCALL SDL_FileTimeToUnix(Uint64 ftime);
```

Comment:
My compiler complained here because it conflicted in `SDL_dynapi_procs.h`.
```suggestion
extern DECLSPEC void SDLCALL SDL_FileTimeToWindows(Sint64 ftime, Uint32 *low, Uint32 *high);

/* converts an SDL file timestamp into a Unix time_t (seconds since the Unix epoch). */
extern DECLSPEC Sint64 SDLCALL SDL_FileTimeToUnix(Uint64 ftime);
```

Reply:
Whoops, I never `git add`ed the header file for some reason. Fixed in a9de3ed62d6d93f71f2628b9df72f363cc0b9d5f!